### PR TITLE
Refactoring the topic config, passing directly in the broker config map

### PR DESCRIPTION
**Pull request by matzew**
Signed-off-by: Matthias Wessendorf <mwessend@redhat.com>

Fixes #

## Proposed Changes

- changing the `topicConfig` implementation and passing in the "broker config" `ConfigMap`

**Release Note**

```release-note

```

**Docs**

### CODE REVIEW

File: control-plane/pkg/reconciler/broker/broker.go [hunk 1]
```diff
@@ -108,7 +108,12 @@ func (r *Reconciler) reconcileKind(ctx context.Context, broker *eventing.Broker)
...
 
-	topicConfig, brokerConfig, err := r.topicConfig(logger, broker)
+	brokerConfig, err := r.brokerConfigMap(logger, broker)
+	if err != nil && !apierrors.IsNotFound(err) {
+		return statusConditionManager.FailedToResolveConfig(err)
+	}
+
+	topicConfig, err := r.topicConfig(logger, broker, brokerConfig)
 	if err != nil {
 		return statusConditionManager.FailedToResolveConfig(err)
```
Comment (matzew) [inline control-plane/pkg/reconciler/broker/broker.go:118]: these two `FailedToResolveConfig` I don't like that much ..., yet ...

Comment (matzew) [reply control-plane/pkg/reconciler/broker/broker.go:118]: it's called more often, on different areas... I am letting this as is...

**Review by pierDipi — APPROVED** [review]
/lgtm
/approve

**Pull request merged**